File: src/main/java/dev/interfiber/karpet/server/recipes/StickRecipe.java
package dev.interfiber.karpet.server.recipes;

import net.minestom.server.item.ItemStack;
import net.minestom.server.item.Material;
import net.minestom.server.network.packet.server.play.DeclareRecipesPacket.Ingredient;

import java.util.List;

public class StickRecipe extends MinecraftRecipe {
    public StickRecipe(){
        this.RecipeID = "stick";
        this.Items = List.of(
                ItemStack.of(Material.OAK_PLANKS, 1),
                ItemStack.AIR,
                ItemStack.of(Material.OAK_PLANKS, 1),
                ItemStack.AIR
        );
        this.Result = ItemStack.of(Material.STICK, 4);
    }
    @Override
    public boolean CanCraftInPortableCrafting(){
        return true;
    }
}

File: src/main/kotlin/dev/interfiber/karpet/server/recipes/StickRecipe.kt
package dev.interfiber.karpet.server.recipes

import net.minestom.server.item.ItemStack
import net.minestom.server.item.Material
/**
 * Recipe to craft a stick
 * @author Interfiber
 */
class StickRecipe : MinecraftRecipe() {
    init {
        this.recipeID = "stick"
        this.portableItems = listOf(
            ItemStack.of(Material.OAK_PLANKS, 1),
            ItemStack.AIR,
            ItemStack.of(Material.OAK_PLANKS, 1),
            ItemStack.AIR
        )
        // anything that is a * is taken
        // * x x <- row one
        // * x x <- row two
        // x x x <- row three
        this.items = listOf(
            ItemStack.of(Material.OAK_PLANKS, 1), // *
            ItemStack.AIR, // x
            ItemStack.AIR, // x
            ItemStack.of(Material.OAK_PLANKS, 1), // *
            ItemStack.AIR, // x
            ItemStack.AIR, // x
            ItemStack.AIR, // x
            ItemStack.AIR, // x
            ItemStack.AIR, // x
        )
        this.result = ItemStack.of(Material.STICK, 4)
    }

    override fun canCraftInPortableCrafting(): Boolean {
        return true
    }
}